### ISSUE TITLE
`plotSPC(..., hz.depths=TRUE)` warnings "data should be pre-sorted", "hzd.txt.y - hzd.txt.y.fixed :  longer object length is not a multiple of shorter object length"

### ISSUE THREAD

**Issue by brownag** [ncss-tech/aqp#290]
The following warning `"data should be pre-sorted"` is issued with `plotSPC(sp1, hz.depths=TRUE)`. It appears the data are sorted already so possibly this is not considering all possible profile/horizon geometries.

``` r
library(aqp)
#> This is aqp 2.0
data(sp1)
depths(sp1) <- id ~ top + bottom
plotSPC(
  sp1,
  cex.names = 1,
  hz.depths = TRUE
)
#> Warning: data should be pre-sorted
```

![](https://i.imgur.com/OBHmamY.png)

If the SoilProfileCollection plotted has one or more horizons have equal top and bottom depth (e.g. sp1 dataset profile 1), then the warning is different: `"hzd.txt.y - hzd.txt.y.fixed :  longer object length is not a multiple of shorter object length"`. In this case the horizon with identical top and bottom depth seems to be labeled twice, once in the correct position and once overlapping one of the other horizons.

``` r
plotSPC(
    sp1[1,],
    cex.names = 1,
    hz.depths = TRUE
)
#> Warning in hzd.txt.y - hzd.txt.y.fixed: longer object length is not a multiple
#> of shorter object length
```

![](https://i.imgur.com/xR6QqLL.png)

**Comment by dylanbeaudette** [comment]
Good catch, this is clearly a bug in `fixOverlap()` but only when called from `plotSPC()` as an un-handled corner case with duplicate horizon depths and `NA` last bottom depth.

**Comment by dylanbeaudette** [comment]
Related TODOs:
- [ ] testing for bottom-most hz top == bottom → invoke `repairMissingHzDepths()`
- [ ] update related documentation for `plotSPC()` and sketches tutorial
- [ ] fix `repairMissingHzDepths()` when calling with single-profile SPC

**Comment by brownag** [comment]
> I'm thinking that `plotSPC()` should not modify data to work around problems, rather warn the user when some functionality will break when there are missing / duplicate horizon depths.

Agreed, for me it often takes that visual cue of something being messed up, or a specific warning/error message, to get me to dig into apparent problems.  If `plotSPC()` auto-fixed/obscured the source of the issue in the data it is harder to use for that type of QC. 

Thanks for triaging; seems there are a couple related things worth addressing. I can probably take a look at needed fixes for `repairMissingHzDepths()` in the not-too-distant future if you don't sort it out

**Comment by dylanbeaudette** [comment]
Fixed in #292. Report and ignore those profiles where at least one horizon top == bottom.
